web/js/utils-0.0.23.js

```diff
@@ -1716,32 +1716,250 @@ function pageReadyMast() {
 }
 
 function domReadyMenu() {
-    var sbox = document.getElementById('sbox');
-/*
-    $("#project").autocomplete(projects, {
-        minChars: 0,
-        multiple: true,
-        multipleSeparator: ",",
-        //mustMatch: true,
-        matchContains: "word",
-        max: 200,
-        cacheLength:20,
-        //autoFill: false,
-        formatItem: function(row, i, max) {
-                return (row != null) ? i + "/" + max + ": " + row[0] : "";
-            },
-        formatMatch: function(row, i, max) {
-                return (row != null) ? row[0] : "";
-            },
-        formatResult: function(row) {
-                return (row != null) ? row[0] : "";
-            },
-        width: "300px"
+    $.ajax({
+        // cannot use "/api/v1/configuration/suggester" because of security
+        url: window.contextPath + "/suggest/config",
+        dataType: "json",
+        success: function(config) {
+            if (config.enabled) {
+                initAutocomplete(config);
+            }
+        },
+        error: function(xhr, ajaxOptions, error) {
+            console.log('Could not get autocomplete configuration, probably disabled');
+        }
     });
-*/
-    // TODO  Bug 11749
-    // var p = document.getElementById('project');
-    // p.setAttribute("autocomplete", "off");
+}
+
+function initAutocomplete(config) {
+    if (config.allowedFields) {
+        if (config.allowedFields.includes('full')) {
+            initAutocompleteForField("q", "full", config);
+        }
+        if (config.allowedFields.includes('defs')) {
+            initAutocompleteForField("defs", "defs", config);
+        }
+        if (config.allowedFields.includes('refs')) {
+            initAutocompleteForField("refs", "refs", config);
+        }
+        if (config.allowedFields.includes('path')) {
+            initAutocompleteForField("path", "path", config);
+        }
+        if (config.allowedFields.includes('hist')) {
+            initAutocompleteForField("hist", "hist", config);
+        }
+    } else {
+        initAutocompleteForField("q", "full", config);
+        initAutocompleteForField("defs", "defs", config);
+        initAutocompleteForField("refs", "refs", config);
+        initAutocompleteForField("path", "path", config);
+        initAutocompleteForField("hist", "hist", config);
+    }
+}
+
+function initAutocompleteForField(inputId, field, config) {
+    var text;
+    var identifier;
+    var time;
+
+    var input = $("#" + inputId);
+
+    input.autocomplete({
+        source: function(request, response) {
+
+            var caretPos = $('#' + inputId).caret();
+            if (!Number.isInteger(caretPos)) {
+                console.error("Suggest: could not get caret position");
+                return;
+            }
+
+            $.ajax({
+                url: window.contextPath + "/suggest",
```

Comment:
does it work in the @vladak setup? with the authorization and all these stuff?

Reply:
Just tried to deploy on the staging server and seems to work fine w.r.t. authorization (request with faked user SSO headers to `http://localhost:8080/source/suggest?full=&field=full&caret=0&projects%5B%5D=proj` fails with 403 where `proj` is project the user does not have access to)